src/app/api/user/[id]/route.ts

```diff
@@ -28,3 +31,54 @@ export async function GET(
     return new Response("Internal Server Error", { status: 500 });
   }
 }
+
+/**
+ * PATCH function that updates a user's details
+ */
+export async function PATCH(
+  req: NextRequest,
+  { params }: { params: { id: string } },
+) {
+  const { id } = params;
+
+  // Validate the body of the request
+  const json = await req.json();
+  const bodyResult = updateUserSchema.safeParse(json);
+
+  if (!bodyResult.success) {
+    return NextResponse.json(bodyResult.error, { status: 400 });
+  }
+
+  const { firstName, lastName, member } = bodyResult.data;
+
+  // Check that the current user is defined
+  const currentUser = await getCurrentUser();
+  if (!currentUser) {
+    return new Response("No current user logged in", { status: 400 });
```

Comment:
return null, status:403 instead
this indicates that the user is unauthorised to access the resource (not logged in)

in this same if statement you should also make sure that currentUser.id is equal to the params.id otherwise return 403

Reply:
Updated this to be a 403